simple_salesforce/api.py

```diff
@@ -267,7 +267,7 @@ def query(self, query, **kwargs):
         * query -- the SOQL query to send to Salesforce, e.g.
                    `SELECT Id FROM Lead WHERE Email = "waldo@somewhere.com"`
         """
-        url = self.base_url + 'query/'
+        url = self.base_url + ( 'queryAll/' if ( 'query_all' in kwargs and kwargs.pop('query_all') ) else 'query/' )
```

Comment:
Not a huge fan of having this all inline (plus it goes past the 80 char pep8 limit)

If we decide to go this route we'll probably need refactor this, then add something to the docstring.